mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,328 @@
+			httpLogs.GetLogEntry()[num].GetCommonProperties().GetTimeToFirstUpstreamRxByte().Nanos)
+		reqMap["request.useragent"] = httpLogs.GetLogEntry()[num].GetRequest().GetUserAgent()
+		reqMap["response.headers"] = attr.WrapStringMap(httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeaders())
+		reqMap["response.code"] = int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseCode().GetValue())
+		reqMap["response.size"] = int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseBodyBytes())
+		reqMap["response.total_size"] = int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseBodyBytes()) +
+			int64(httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeadersBytes())
+		reqMap["UPSTREAM_CLUSTER"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetUpstreamCluster()
+		reqMap["connection.requested_server_name"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetTlsProperties().GetTlsSniHostname()
+		reqMap["context.proxy_error_code"] = ParseEnvoyResponseFlags(httpLogs.GetLogEntry()[num].GetCommonProperties().GetResponseFlags())
+		fillContextProtocol(reqMap)
+	} else if tcpLogs := msg.GetTcpLogs(); tcpLogs != nil {
+		reqMap["context.reporter.kind"] = "inbound"
+		fillAddress(reqMap, tcpLogs.GetLogEntry()[num].GetCommonProperties().GetDownstreamLocalAddress(), "destination")
+		fillAddress(reqMap, tcpLogs.GetLogEntry()[num].GetCommonProperties().GetDownstreamRemoteAddress(), "source")
+		reqMap["request.time"] = reformatTime(tcpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime(), 0)
+		reqMap["response.time"] = reformatTime(tcpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime(),
+			tcpLogs.GetLogEntry()[num].GetCommonProperties().GetTimeToFirstUpstreamRxByte().Nanos)
+		reqMap["connection.requested_server_name"] = tcpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetTlsProperties().GetTlsSniHostname()
+		reqMap["connection.received.bytes"] = tcpLogs.GetLogEntry()[num].GetConnectionProperties().GetReceivedBytes()
+		reqMap["connection.sent.bytes"] = tcpLogs.GetLogEntry()[num].GetConnectionProperties().GetSentBytes()
+		reqMap["UPSTREAM_CLUSTER"] = tcpLogs.GetLogEntry()[num].GetCommonProperties().GetUpstreamCluster()
+		reqMap["context.proxy_error_code"] = ParseEnvoyResponseFlags(tcpLogs.GetLogEntry()[num].GetCommonProperties().GetResponseFlags())
+		reqMap["context.protocol"] = "tcp"
+	}
+
+	pb.reqMap = reqMap
+	scope.Debugf("Returning bag with attributes:\n%v", pb)
+	return pb
+}
+
+//fills in destination.service.name and destination.service.host after the initial bag has been built
+func (pb *EnvoyProtoBag) AddNamespaceDependentAttributes(destinationNamespace string) {
+	upstreamCluster := pb.reqMap["UPSTREAM_CLUSTER"]
+	parts := strings.Split(fmt.Sprintf("%v", upstreamCluster), "|")
+	var host string
+	if len(parts) == 4 {
+		host = parts[3]
+	} else {
+		host = fmt.Sprintf("%v", pb.reqMap["request.host"])
```

Comment:
fmt.Sprintf is not efficient. you can just use type assertion here. pb.reqMap["request.host"].(string)